pygmt/src/meca.py

```diff
@@ -442,20 +435,19 @@ def update_pointers(data_pointers):
         else:
             raise GMTError("Parameter 'spec' contains values of an unsupported type.")
 
+    # Ensure non-file types are a 2d array
+    if isinstance(spec, (list, np.ndarray)):
+        spec = np.atleast_2d(spec)
```

Comment:
This is a brute-force solution to make 1d lists and numpy.arrays work (c.f. #1132). Note to self to remove this, because it will break if we allow for text labels (i.e. `str` dtype) on beachballs later.